Services/Repository.cs

```diff
@@ -0,0 +1,22 @@
+using System.Linq.Expressions;
+
+
+namespace TodoNET6.Services
+{
+    public interface IEntity<TIndex>
+    {
+        TIndex Id { get; set; }
+    }
+
+    public interface IRepository<TEntity, TIndex>
+        where TEntity : IEntity<TIndex>
+    {
+        Task<TEntity> CreateAsync(TEntity entity);
+        Task<TEntity?> FindAsync(params object[] values);
```

Comment:
A `params` value is how you tell the compiler that the array passed in is of variable length and gives you syntactic sugar to allow you to define the parameters of the function like:

```CSharp
FindAsync(val1, val2, val3)
```

rather than
```CSharp
FindAsync(new object[] { val1, val2, val3 });
```